x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/TransformTaskParams.java

```diff
@@ -50,28 +54,55 @@ public class TransformTaskParams implements SimpleDiffable<TransformTaskParams>,
         PARSER.declareLong(ConstructingObjectParser.optionalConstructorArg(), FROM);
         PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), FREQUENCY);
         PARSER.declareBoolean(ConstructingObjectParser.optionalConstructorArg(), REQUIRES_REMOTE);
+        PARSER.declareBoolean(ConstructingObjectParser.optionalConstructorArg(), UNATTENDED);
     }
 
-    private TransformTaskParams(String transformId, String version, Long from, String frequency, Boolean remote) {
+    private TransformTaskParams(String transformId, String version, Long from, String frequency, Boolean remote, Boolean unattended) {
         this(
             transformId,
             version == null ? null : TransformConfigVersion.fromString(version),
             from == null ? null : Instant.ofEpochMilli(from),
             frequency == null ? null : TimeValue.parseTimeValue(frequency, FREQUENCY.getPreferredName()),
-            remote == null ? false : remote.booleanValue()
+            remote != null && remote,
+            unattended
         );
     }
 
-    public TransformTaskParams(String transformId, TransformConfigVersion version, TimeValue frequency, boolean remote) {
-        this(transformId, version, null, frequency, remote);
-    }
-
-    public TransformTaskParams(String transformId, TransformConfigVersion version, Instant from, TimeValue frequency, boolean remote) {
+    public TransformTaskParams(
+        String transformId,
+        TransformConfigVersion version,
+        TimeValue frequency,
+        boolean remote,
+        boolean unattended
+    ) {
+        this(transformId, version, null, frequency, remote, unattended);
+    }
+
+    public TransformTaskParams(
+        String transformId,
+        TransformConfigVersion version,
+        Instant from,
+        TimeValue frequency,
+        boolean remote,
+        boolean unattended
+    ) {
+        this(transformId, version, from, frequency, remote, Boolean.valueOf(unattended));
+    }
+
+    private TransformTaskParams(
```

Comment:
This is intentionally private and overloaded with the public constructor.  I'd like `unattended` to be a primitive `boolean`, but that isn't possible until every node in the cluster supports this object (I think?). So anyone compiling with this code should use `boolean`, and anyone transporting the object over the wire can use `Boolean`.

Reply:
> (I think?)

Even when every node in the cluster supports `unattended` in the task parameters it's still possible that some transforms won't have it in their task parameters. A transform could be started in 8.12 and never, ever, be stopped. It would hop nodes through every rolling restart and still have the same task parameters while eventually running in version 11.6 in ten years time. Very unlikely I know, but possible.

Reply:
> This is intentionally private and overloaded with the public constructor.

I don't we really need this constructor. We already have a constructor that deals with `unattended` being `null` (line 60) so we could do the `Boolean` to `boolean` conversion there.

BTW, I'm wondering about updates. If it is possible to update the transform from unattended to non-unattended and vice-versa then we would need to effectively update task parameters which is AFAIK not possible. We can chat about it offline.